src/cc65/datatype.c

```diff
@@ -245,6 +274,49 @@ const Type* GetStructReplacementType (const Type* SType)
 
 
 
+long GetIntegerTypeMin (const Type* Type)
+/* Get the smallest possible value of the integer type */
+{
+    if (IsSignSigned (Type)) {
+        return (long)(0xFFFFFFFF << (CHAR_BITS * SizeOf (Type) - 1U));
```

Comment:
Maybe it's just me being too stupid, but it seems to me that this line deserves a comment how it works.

Apart from that I don't understand why it's not appropriate to use `LONG_MAX` here - given that `long` is used. Maybe that would be worth added to a comment as well.

Reply:
The min possible signed value of the byte size N is `- pow (2, 8*N-1)`. I'll add the comment.

And I am not sure why you refer to `LONG_MAX` here, given that this funtion is used for calcaulating the `MIN` rather than the `MAX` for an arbitray integer type....

Reply:
If you were questioning about why the check for `unsigned long` overflow doesn't simply compare the enumerator value to `ULONG_MAX` before the incrementation... well, I did that in earlier revisions but turned to the current method as I'd like just to warn when the "overflow has already happened", although that may just be unconvincing.

Reply:
Sorry for being too unprecise. In fact I was only wondering if `0xFFFFFFFF` couldn't be replaced by something from `limits.h`.

Reply:
> 
> 
> Sorry for being too unprecise. In fact I was only wondering if `0xFFFFFFFF` couldn't be replaced by something from `limits.h`.

The number used here is supposed to be -1, but due to quirks with signed integer shift I decided to use explicit hex literal (which is always unsigned) for it. It could be changed to `(long)((unsigned long)(-1L) << (CHAR_BITS * SizeOf (Type) - 1U))` if that's considered more appropriate.

Reply:
Thanks, at least I understand it better now :-)